include/ccf/service/tables/jwt.h

```diff
@@ -58,18 +58,28 @@ namespace ccf
   using JwtKeyId = std::string;
   using Cert = std::vector<uint8_t>;
 
+  struct JwtIssuerWithConstraint
+  {
+    JwtIssuer issuer;
+    std::optional<JwtIssuer> constraint;
+  };
+  DECLARE_JSON_TYPE_WITH_OPTIONAL_FIELDS(JwtIssuerWithConstraint);
+  DECLARE_JSON_REQUIRED_FIELDS(JwtIssuerWithConstraint, issuer);
+  DECLARE_JSON_OPTIONAL_FIELDS(JwtIssuerWithConstraint, constraint);
+
   using JwtIssuers = ServiceMap<JwtIssuer, JwtIssuerMetadata>;
   using JwtPublicSigningKeys = kv::RawCopySerialisedMap<JwtKeyId, Cert>;
-  using JwtPublicSigningKeyIssuer =
```

Comment:
Spoke with @achamayou that it's not safe for upgrade. We'll proceed with keeping the old table in this version, and will get rid of it later in subsequent 5.0.x release.

Reply:
Change of plans. We'll do

* Keep old tables to fallback to if we have no keys in the new tables to keep serving requests
* Don't do anything to the old code. This might cause stale keys which we are already forced to have due to 30-min refresh window, so shouldn't make things noticeably worse
* The latter is bad only if we remove the key on the new node and old nodes keep serving requests with the old keys, which isn't good, but almost unachievable in practise.

Reply:
Done